client/src/main/scala/com/dwolla/cloudflare/clients/AccountsClient.scala

```diff
@@ -0,0 +1,126 @@
+package com.dwolla.cloudflare.clients
+
+import cats._
+import cats.implicits._
+import com.dwolla.cloudflare.CloudflareApiExecutor
+import com.dwolla.cloudflare.common.JsonEntity._
+import com.dwolla.cloudflare.common.UriHelper
+import com.dwolla.cloudflare.domain.dto.accounts.{AccountDTO, AccountMemberDTO, AccountRoleDTO, NewAccountMemberDTO}
+import com.dwolla.cloudflare.domain.dto.{PagedResponseDTO, ResponseDTO}
+import com.dwolla.cloudflare.domain.model.DeletedRecord
+import com.dwolla.cloudflare.domain.model.Exceptions.UnexpectedCloudflareErrorException
+import com.dwolla.cloudflare.domain.model.accounts.Implicits._
+import com.dwolla.cloudflare.domain.model.accounts.{Account, AccountMember, AccountRole}
+import com.dwolla.cloudflare.domain.model.response.Implicits._
+import com.dwolla.cloudflare.domain.model.response.PagedResponse
+import org.apache.http.client.methods._
+import org.json4s.native._
+import org.json4s.{DefaultFormats, Formats}
+
+import scala.language.higherKinds
+
+class AccountsClient[F[_] : Monad](executor: CloudflareApiExecutor[F]) {
+  protected implicit val formats: Formats = DefaultFormats
+
+  def list(page: Int = 1, perPage: Int = 25): F[PagedResponse[Set[Account]]] = {
+    val parameters = UriHelper.buildParameterString(Seq(Option("page" → page), Option("per_page" → perPage), Option("direction" → "asc")))
+    val request: HttpGet = new HttpGet(UriHelper.buildApiUri("accounts", Some(parameters)))
+
+    executor.fetch(request) { response ⇒
+      parseJson(response.getEntity.getContent).extract[PagedResponseDTO[Set[AccountDTO]]]
+    }
+  }
+
+  def listAll(): F[Set[Account]] = {
+    list(1).flatMap { pagedResponse ⇒
```

Comment:
Will this go fetch the entire list (across all pages) when this function is called?

If so, fetching unlimited data without a consumer that actually requires it concerns me a little bit. I wonder if it would be better to switch this to a stream-based interface, e.g. using fs2 and returning `Stream[F, Account]`?

Reply:
This will fetch all pages. I was using it to find an account by name, since that is not supported by the Cloudflare API. Having more that 25 accounts seems unlikely, but it could happen.